src/WebJobs.Extensions.DurableTask/ContextInterfaces/IDurableOrchestrationClient.cs

```diff
@@ -261,6 +261,16 @@ public interface IDurableOrchestrationClient
         /// <returns>Returns a task which completes when the status has been fetched.</returns>
         Task<DurableOrchestrationStatus> GetStatusAsync(string instanceId, bool showHistory = false, bool showHistoryOutput = false, bool showInput = true);
 
+        /// <summary>
+        /// Get the status of multiple instances.
+        /// </summary>
+        /// <param name="instanceIds"> The instanceIDs to query.</param>
+        /// <param name="showHistory">Boolean marker for including execution history in the response.</param>
+        /// <param name="showHistoryOutput">Boolean marker for including input and output in the execution history response.</param>
+        /// <param name="showInput">If set, fetch and return the input for the orchestration instance. If both <c>showHistory</c> and <see cref = "DurableTaskOptions.StoreInputsInOrchestrationHistory" /> are also set to<c>true</c>, then the inputs for activity and sub-orchestration events in the orchestration history will also be returned.</param>
+        /// <returns>Returns a list of orchestration statuses.</returns>
+        Task<IList<DurableOrchestrationStatus>> GetStatusAsync(IEnumerable<string> instanceIds, bool showHistory = false, bool showHistoryOutput = false, bool showInput = false);
```

Comment:
For methods that have a large list of individual "options" parameters, I like to wrap them into a record class. This is beneficial for a couple reasons:

1. You can add more options without making a signature change (which btw, is a breaking change on interfaces!)
2. Having a class makes it easy to store/re-use their query.
3. `record`'s with their immutability and `with` parameter are perfect for this scenario, as you can transform a shared query without impacting other uses of it.

To summarize, consider something like:

``` CSharp
IAsyncEnumerable<DurableOrchestrationStatus> GetStatusAsync(IEnumerable<string> instanceIds, GetStatusOptions? options = null);

public record GetStatusOptions(bool ShowHistory = false, bool ShowHistoryOutput = false, ShowInput = false);
```